File: Computing1/Past_Computing1/22-1_컴기_Homework/Homework/HW12/P1.py
def P1(nums: list, k: int) -> bool:

    ### Write code here ###
    dct = dict()
    
    for  i in nums:
        if i% k  in dct:
            dct[i%k] += 1
        else:
            dct[i%k] = 1
    
    for i in dct:
        if i == 0 or i * 2 == k:
            if dct[i] % 2 != 0:
                return False
#             else:
#                 return True
        else:
            if k-i not in dct:
                return False
            if dct[i] != dct[k-i]: 
                return False
    return True
    
    

    
    ### End of your code ###v

File: Computing1/Past_Computing1/22-1_컴기_Homework/Homework/HW20_v3/P1.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]){
    int N = atoi(argv[1]);
    // YOUR CODE HERE
    if (N == 1){printf("*");}
    else if (N == 2){printf(" * \n***");}
    else {
        int i, j;

        for (i=1;i<=(N-1); i++){printf(" ");}
        printf("*"); // 0, 1, 1+2, 1+2+2, 1+2+2+2,
        for (i=1;i<=(N-1); i++){printf(" ");}
        printf("\n");
        
        for (j=2; j<N; j++){
            for (i=N-j; i>=1; i--){printf(" ");} // N-2 N-3 --- 1
            printf("*");
            for (i=2*N-(1+2*j); i<=(2*N-5); i++){printf(" ");} // 2N-19, 2N-17, 2N-15,..., 2N-5
            printf("*");
            printf("\n");
        }

        for (i=0;i<(2*N-1); i++){printf("*");}
    }
    return 0;
}
